#!/usr/bin/env python
PACKAGE = "foresight"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("yaw_setpoint", double_t, 0, "Yaw Setpoint", 0.0, 0.0, 7)
gen.add("z_setpoint",   double_t, 0, "Z Setpoint", 1.0, 0.1, 4.0)
gen.add("y_setpoint",   double_t, 0, "Y Setpoint", 1.0, 0.1, 4.0)
gen.add("x_setpoint",   double_t, 0, "X Setpoint", 1.0, 0.1, 4.0)
gen.add("yaw_gains_P",  double_t, 0, "Yaw kp", 0.05, 0, 50)
gen.add("yaw_gains_I",  double_t, 0, "Yaw ki", 0.00, 0, 0.5)
gen.add("yaw_gains_D",  double_t, 0, "Yaw kd", 0.01, 0, 0.5)
gen.add("y_gains_P",  double_t, 0, "Y kp", 0.05, 0, 1)
gen.add("y_gains_I",  double_t, 0, "Y ki", 0.00, 0, 0.5)
gen.add("y_gains_D",  double_t, 0, "Y kd", 0.01, 0, 0.5)
gen.add("x_gains_P",  double_t, 0, "X kp", 0.05, 0, 1)
gen.add("x_gains_I",  double_t, 0, "X ki", 0.00, 0, 0.5)
gen.add("x_gains_D",  double_t, 0, "X kd", 0.01, 0, 0.5)
gen.add("z_gains_P",    double_t, 0, "Z kp", 0.05, 0, 0.5)
gen.add("z_gains_I",    double_t, 0, "Z ki", 0.00, 0, 0.5)
gen.add("z_gains_D",    double_t, 0, "Z kd", 0.01, 0, 0.5)
gen.add("yaw_feedforward", double_t, 0, "Yaw Feedforward", 0.0, 0.0, 0.5)
gen.add("z_feedforward",   double_t, 0, "Z Feedforward", 0.05, 0.0, 0.5)
gen.add("y_feedforward", double_t, 0, "Y Feedforward", 0.0, 0.0, 0.5)
gen.add("x_feedforward",   double_t, 0, "X Feedforward", 0.0, 0.0, 0.5)

exit(gen.generate(PACKAGE, "pd_controller", "PID"))
